<launch>
  <!--
          NUM:=0|1|...
          VISUALIZE:=true|false  (Open cv2 window)
  -->
  
  <arg name="NUM" default="0"/>
  <arg name="SERIAL" default="D20387"/>
  <!-- Visualization with rviz, only if arg VISUALIZE is set to true. Default is
       false, the allegro_viz.launch can be started separated. -->
  <arg name ="VISUALIZE" default="false" />
  <arg name="GPU" default="false" />
  <arg name="SCALE" default="1.0" />
  <arg name="MMPP" default="0.0487334006" />
  <arg name="COMPRESS" default="true" />

  <!-- Allegro Hand controller and communication node. -->
  <node name="digit_$(arg NUM)"
        pkg="digit"
        type="digit_sensor_pub.py"
        output="screen">

    <!-- Remapping of topics into enumerated namespace -->
    <remap from="digit_sensor/raw" to="digit_sensor/$(arg NUM)/raw" unless="$(arg COMPRESS)"/>
    <remap from="digit_sensor/diff" to="digit_sensor/$(arg NUM)/diff" unless="$(arg COMPRESS)"/>
    <remap from="digit_sensor/raw/compressed" to="digit_sensor/$(arg NUM)/raw/compressed" if="$(arg COMPRESS)"/>
    <remap from="digit_sensor/diff/compressed" to="digit_sensor/$(arg NUM)/diff/compressed" if="$(arg COMPRESS)"/>
    <remap from="digit_sensor/depth" to="digit_sensor/$(arg NUM)/depth"/>

    <param name="model_path" value="$(find digit)/data/digit_nn_normalized"/>
    <param name="serial" value="$(arg SERIAL)"/>
    <param name="use_gpu" value="$(arg GPU)"/>
    <param name="scale" value="$(arg SCALE)"/>
    <param name="mmpp" value="$(arg MMPP)"/>
    <param name="visualize" value="$(arg VISUALIZE)"/>
  </node>

</launch>